picasso/src/main/java/com/squareup/picasso3/SourceBufferingInputStream.java

```diff
@@ -79,7 +97,8 @@ private int copyTo(byte[] sink, int offset, int byteCount) {
   }
 
   @Override public void mark(int readlimit) {
-    mark = position + readlimit;
```

Comment:
This doesn't conform to the `InputStream.mark` contract, which states that the readLimit is how far you can read **from** the mark.  This treats the readlimit as the actual mark to return to.